draft-ietf-jsonpath-base.md

```diff
@@ -439,12 +439,25 @@ errors.)
 The JSON value the JSONPath query is applied to is, by definition, a valid JSON value.
 The parsing of a JSON text into a JSON value and what happens if a JSON
 text does not represent valid JSON are not defined by this specification.
-
-When a JSONPath query is applied to a JSON value containing an object with duplicate
-names, the behaviour is unpredictable. (Such a value usually originates
-in JSON text containing an object with duplicate keys, but how such a value could
-be constructed depends on how the JSON text is parsed and how the parsed value is
-represented.)
+{{Sections 4 and 8 of -json}} identify specific situations that may
+conform to the grammar for JSON texts but are not interoperable uses
+of JSON, for instance as they may cause unpredictable behavior.
+The present specification does not attempt to define predictable
+behavior for JSONPath queries in these situations.
+(Note that another warning about interoperability, in {{Section 2 of
+-json}}, at the time of writing is generally considered to be overtaken
+by events and causes no issues with the present specification.)
+
+Specifically, the "Semantics" subsections of Sections {{<dot-selector}},
+{{<wildcard}}, {{<index-selector}}, {{<index-wildcard-selector}},
+{{<filter-selector}}, and {{<descendant-selectors}} describe behavior that
+turns unpredictable when the JSON value for one of the objects
+under consideration was constructed out of JSON text that exhibits
+multiple members for a single object that share the same member name
+("duplicate names", see {{Section 4 of -json}}), and matching ({{selectors}}) and comparing
+({{comparisons}} in Section {{<filter-selector}}) of strings assumes these
+strings are sequences of Unicode scalar values, turning unpredictable
+if they aren't ({{Section 8.2 of -json}}).
```

Comment:
I'm finding it hard to parse this sentence from "and matching" onwards. I suggest splitting into two or more sentences.

Reply:
(see suggestion above)